src/workerd/api/crypto/spkac.c++

```diff
@@ -0,0 +1,95 @@
+#include "spkac.h"
+#include "impl.h"
+#include <workerd/jsg/jsg.h>
+#include <openssl/x509.h>
+#include <openssl/pem.h>
+
+namespace workerd::api {
+namespace {
+kj::ArrayPtr<const kj::byte> trim(kj::ArrayPtr<const kj::byte> input) {
+  size_t length = input.size();
+  for (auto i = length - 1; i >= 0; --i) {
+    if (input[i] != ' ' && input[i] != '\n' && input[i] != '\r' && input[i] != '\t') {
+      break;
+    }
+  }
+  return input.first(length);
+}
+
+kj::Array<kj::byte> toArray(BIO* bio) {
+  BUF_MEM* bptr;
+  BIO_get_mem_ptr(bio, &bptr);
+  auto buf = kj::heapArray<char>(bptr->length);
+  auto aptr = kj::arrayPtr(bptr->data, bptr->length);
+  buf.asPtr().copyFrom(aptr);
+  return buf.releaseAsBytes();
+}
+
+kj::Maybe<kj::Own<NETSCAPE_SPKI>> tryGetSpki(kj::ArrayPtr<const kj::byte> input) {
+  static constexpr int32_t kMaxLength = kj::maxValue;
+  JSG_REQUIRE(input.size() <= kMaxLength, RangeError, "spkac is too large");
+  input = trim(input);
+  auto ptr = NETSCAPE_SPKI_b64_decode(input.asChars().begin(), input.size());
+  if (!ptr) return kj::none;
+  return kj::disposeWith<NETSCAPE_SPKI_free>(ptr);
+}
+
+kj::Maybe<kj::Own<EVP_PKEY>> tryOwnPkey(kj::Own<NETSCAPE_SPKI>& spki) {
+  auto pkey = NETSCAPE_SPKI_get_pubkey(spki.get());
+  if (!pkey) return kj::none;
+  return kj::disposeWith<EVP_PKEY_free>(pkey);
+}
+
+kj::Maybe<kj::Own<BIO>> tryNewBio() {
+  auto bioptr = BIO_new(BIO_s_mem());
+  if (!bioptr) return kj::none;
+  return kj::disposeWith<BIO_free_all>(bioptr);
+}
+}  // namespace
+
+bool verifySpkac(kj::ArrayPtr<const kj::byte> input) {
+  // So, this is fun. SPKAC uses MD5 as the digest algorithm. This is a problem because
```

Comment:
If we can't verify spkac, should we even ship the method? It would be confusing to developers if they can call a method but it only produces errors. We also can't test it reliably.

Reply:
Well, we do have the option of manually implementing it rather than relying on boringssl's implementation. It's a fair question tho and the answer is: I'm not sure.

Reply:
For now I've added a `logWarningOnce(...)` that will explain the situation if the method is called from within an active IoContext. I think this is good enough and we can revisit this later if folks give any pushback.